File: src/Muflone/Messages/IMessageHandlerAsync.cs
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Muflone.Messages;

public interface IMessageHandlerAsync : IDisposable
{
}

public interface IMessageHandlerAsync<TMessage> : IDisposable
    where TMessage : IMessage
{
    Task HandleAsync(TMessage message, CancellationToken cancellationToken = default);
}

File: php/src/Muflone/Messages/IMessageHandlerAsync.php
<?php

declare(strict_types=1);

namespace Muflone\Messages;

interface IMessageHandlerAsync
{
    /**
     * Handle the message asynchronously
     *
     * @param IMessage $message
     * @return void
     */
    public function handleAsync(IMessage $message): void;
}
